lib/runtime/version.go

```diff
@@ -65,131 +61,81 @@ func NewLegacyVersionData(specName, implName []byte, authoringVersion, specVersi
 }
 
 // SpecName returns the spec name
-func (v *LegacyVersionData) SpecName() []byte {
-	return v.specName
+func (lvd *LegacyVersionData) SpecName() []byte {
+	return lvd.specName
 }
 
 // ImplName returns the implementation name
-func (v *LegacyVersionData) ImplName() []byte {
-	return v.implName
+func (lvd *LegacyVersionData) ImplName() []byte {
+	return lvd.implName
 }
 
 // AuthoringVersion returns the authoring version
-func (v *LegacyVersionData) AuthoringVersion() uint32 {
-	return v.authoringVersion
+func (lvd *LegacyVersionData) AuthoringVersion() uint32 {
+	return lvd.authoringVersion
 }
 
 // SpecVersion returns the spec version
-func (v *LegacyVersionData) SpecVersion() uint32 {
-	return v.specVersion
+func (lvd *LegacyVersionData) SpecVersion() uint32 {
+	return lvd.specVersion
 }
 
 // ImplVersion returns the implementation version
-func (v *LegacyVersionData) ImplVersion() uint32 {
-	return v.implVersion
+func (lvd *LegacyVersionData) ImplVersion() uint32 {
+	return lvd.implVersion
 }
 
 // APIItems returns the API items
-func (v *LegacyVersionData) APIItems() []*APIItem {
-	return v.apiItems
+func (lvd *LegacyVersionData) APIItems() []APIItem {
+	return lvd.apiItems
 }
 
 // TransactionVersion returns the transaction version
-func (v *LegacyVersionData) TransactionVersion() uint32 {
+func (lvd *LegacyVersionData) TransactionVersion() uint32 {
 	return 0
 }
 
-// Encode returns the SCALE encoding of the Version
-func (v *LegacyVersionData) Encode() ([]byte, error) {
-	info := &struct {
-		SpecName         []byte
-		ImplName         []byte
-		AuthoringVersion uint32
-		SpecVersion      uint32
-		ImplVersion      uint32
-	}{
-		SpecName:         v.specName,
-		ImplName:         v.implName,
-		AuthoringVersion: v.authoringVersion,
-		SpecVersion:      v.specVersion,
-		ImplVersion:      v.implVersion,
-	}
+type legacyVersionData struct {
```

Comment:
What is the purpose of having both this stuct and LegacyVersionData?

Reply:
it looks like  this has exported fields so its used to decode with SCALE?

Reply:
Yea the public struct implements an interface with getters and setters, which conflict with the attribute names if I were to make them public.  Generics will be nice to implement this.